### PR TITLE
JOSS Paper

### PR DESCRIPTION
**Pull request by bpatel2107**
PR to store the initial JOSS paper submission

Compiled paper can found under GithubActions -> Artefacts -> paper

Info on the JOSS submission requirements can be found here 
https://joss.readthedocs.io/en/latest/submitting.html

### CODE REVIEW

File: docs/joss/paper.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+mitigate turbulent transport and thus requires high levels of confidence in the tools being
+used. Many different gyrokinetic modelling codes are available for use and pyrokinetics aims to 
+standardise the analysis of such simulations.
+
+# Statement of need
+
+[Pyrokinetics](https://github.com/pyro-kinetics/pyrokinetics) is a Python project (package: 
+[`pyrokinetics`](https://pypi.org/project/pyrokinetics/))
+that aims to simplify and standardise gyrokinetic analysis. A wide 
+variety of different gyrokinetic codes exist that utilise different input files and
```
Comment (ZedThree) [inline docs/joss/paper.md]:
```suggestion
variety of different gyrokinetic solvers exist that utilise different input file formats and
```

File: docs/joss/paper.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+use different normalisations for plasma parameters such as densities, temperatures, velocities,
+and magnetic fields. To gain confidence in the prediction of gyrokinetic codes it is often
```
Comment (ZedThree) [inline docs/joss/paper.md]:
```suggestion
and magnetic fields. To improve confidence in the predictions from gyrokinetic solvers it is often 
```

File: docs/joss/paper.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+[Pyrokinetics](https://github.com/pyro-kinetics/pyrokinetics) is a Python project (package: 
+[`pyrokinetics`](https://pypi.org/project/pyrokinetics/))
+that aims to simplify and standardise gyrokinetic analysis. A wide 
+variety of different gyrokinetic codes exist that utilise different input files and
+use different normalisations for plasma parameters such as densities, temperatures, velocities,
+and magnetic fields. To gain confidence in the prediction of gyrokinetic codes it is often 
+desirable to benchmark the results of one code against another. Pyrokinetics aims to ensure that
+all physical plasma parameters given as inputs to gyrokinetic codes are accurate,
+acting as an interface between different codes. Furthermore, gyrokinetics inputs can come from a
```
Comment (ZedThree) [inline docs/joss/paper.md]: I think it's important to clarify that we don't _check_ the inputs, just automagically convert between normalisations etc. Maybe:

```suggestion
desirable to benchmark the results of one code against another. Pyrokinetics aims to make this
easier for researchers by acting as an interface between the different codes, automatically 
handling the conversion of physical input parameters between different normalisations
and file formats. Furthermore, gyrokinetics inputs can come from a
```

File: docs/joss/paper.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
```
Comment (ZedThree) [inline docs/joss/paper.md]: Might be good to give a couple of examples of the other tools we can read inputs from?

File: docs/joss/paper.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and 
+has been built in a modular way such that it is trivial to add in new sources of data.
```
Comment (ZedThree) [inline docs/joss/paper.md]: Not just modular, but explicitly designed to be extended!

```suggestion
has been designed to be extensible and easy to add new sources of data. 
```

File: docs/joss/paper.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+and magnetic fields. To gain confidence in the prediction of gyrokinetic codes it is often 
+desirable to benchmark the results of one code against another. Pyrokinetics aims to ensure that
+all physical plasma parameters given as inputs to gyrokinetic codes are accurate,
+acting as an interface between different codes. Furthermore, gyrokinetics inputs can come from a
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and 
+has been built in a modular way such that it is trivial to add in new sources of data. 
+
+The output of gyrokinetic codes is often multidimensional and each code stores this data in a
+different format with different normalisations. Pyrokinetics will read in all this data and
```
Comment (ZedThree) [inline docs/joss/paper.md]:
```suggestion
different format with different normalisations, and potentially across multiple files. Pyrokinetics will seamlessly read in all this data and
```

File: docs/joss/paper.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
+standard normalisation (using [`pint`](https://pypi.org/project/Pint/)), allowing for direct comparisons between codes. Furthermore, additional derived
+outputs such as the linear growth rate of a turbulent instability, can be calculated using the exact
+same method such that the modeller is confident that the output is consistent across codes.
+
+Pyrokinetics was designed to be used by gyrokinetics modellers and has already been used in a 
+number of scientific publications 
+[@giacomin:2023a; @giacomin:2023b; @kennedy:2023]. Furthermore, the 
+interface with python opens up gyrokinetic analysis to the wide variety of python packages available,
```
Comment (ZedThree) [inline docs/joss/paper.md]:
```suggestion
python interface opens up gyrokinetic analysis to the wide variety of python packages available, 
```

File: docs/joss/paper.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+further interfacing with the whole fusion community and the potential to develop a global gyrokinetic
+database.
+
+We hope that pyrokinetics makes gyrokinetics modelling more accessible and will help to increase the
+communities confidence in tools available.
+
+
+# Acknowledgements
+
+We acknowledge contributions from PlasmaFAIR
```
Comment (ZedThree) [inline docs/joss/paper.md]:
```suggestion
We acknowledge contributions from [PlasmaFAIR](https://plasmafair.github.io), EPSRC Grant EP/V051822/1
```

**Review by ZedThree — COMMENTED** [review]
Looks good! I've made some suggestions for tightening up the language a little

File: docs/joss/paper.md (hunk 4 above)
Comment (bpatel2107) [reply docs/joss/paper.md]: Do you mean like TRANSP, JETTO, GEQDSK etc.?

Comment (bpatel2107) [reply docs/joss/paper.md]: I guess I meant to say from codes like TRANSP/JETTO or formats like GEQDSK?

File: docs/joss/paper.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+   index: 3
+date: 03 August 2023
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Fusion energy offers the potential for a near limitless source of low carbon energy and is often 
+regarded as a solution for the worlds long term energy needs. To realise such a scenario requires
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
regarded as a solution for the world's long term energy needs. To realise such a scenario requires
```

File: docs/joss/paper.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+the design of high performance fusion reactors capable of containing the extreme conditions
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
the design of high performance fusion reactors capable of maintaining the extreme conditions
```

File: docs/joss/paper.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+[@giacomin:2023a; @giacomin:2023b; @kennedy:2023]. Furthermore, the 
+python interface opens up gyrokinetic analysis to the wide variety of python packages available, 
+allowing for a range of analyses from simple parameter scans to using 
+thousands of linear gyrokinetic runs to develop Gaussian process regression models of the
+linear properties of electromagnetic turbulence [@hornsby:2023]. Pyrokinetics also maintains 
+compatibility with IMAS, a community wide standard for data format [@imbeaux:2015], allowing for 
+further interfacing with the whole fusion community and the potential to develop a global gyrokinetic
+database.
+
+We hope that pyrokinetics makes gyrokinetics modelling more accessible and will help to increase the
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
With pyrokinetics we strive to make gyrokinetic modelling more accessible and to increase the
```

File: docs/joss/paper.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+communities confidence in tools available.
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
community's confidence in the tools available.
```

File: docs/joss/paper.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+has been designed to be extensible and easy to add new sources of data. 
+
+The output of gyrokinetic codes is often multidimensional and each code stores this data in a
+different format with different normalisations, and potentially across multiple files. Pyrokinetics will seamlessly read in all this data and
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
+standard normalisation (using [`pint`](https://pypi.org/project/Pint/)), allowing for direct comparisons between codes. Furthermore, additional derived
+outputs such as the linear growth rate of a turbulent instability, can be calculated using the exact
+same method such that the modeller is confident that the output is consistent across codes.
+
+Pyrokinetics was designed to be used by gyrokinetics modellers and has already been used in a
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
Pyrokinetics is designed to be used by gyrokinetics modellers and has already been used in several
```

File: docs/joss/paper.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+The output of gyrokinetic codes is often multidimensional and each code stores this data in a
+different format with different normalisations, and potentially across multiple files. Pyrokinetics will seamlessly read in all this data and
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
+standard normalisation (using [`pint`](https://pypi.org/project/Pint/)), allowing for direct comparisons between codes. Furthermore, additional derived
+outputs such as the linear growth rate of a turbulent instability, can be calculated using the exact
+same method such that the modeller is confident that the output is consistent across codes.
+
+Pyrokinetics was designed to be used by gyrokinetics modellers and has already been used in a 
+number of scientific publications
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
scientific publications 
```

File: docs/joss/paper.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
+standard normalisation (using [`pint`](https://pypi.org/project/Pint/)), allowing for direct comparisons between codes. Furthermore, additional derived
+outputs such as the linear growth rate of a turbulent instability, can be calculated using the exact
+same method such that the modeller is confident that the output is consistent across codes.
+
+Pyrokinetics was designed to be used by gyrokinetics modellers and has already been used in a 
+number of scientific publications 
+[@giacomin:2023a; @giacomin:2023b; @kennedy:2023]. Furthermore, the 
+python interface opens up gyrokinetic analysis to the wide variety of python packages available, 
+allowing for a range of analyses from simple parameter scans to using
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
allowing for a range of analyses from simple parameter scans to the use of
```

File: docs/joss/paper.md [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+same method such that the modeller is confident that the output is consistent across codes.
+
+Pyrokinetics was designed to be used by gyrokinetics modellers and has already been used in a 
+number of scientific publications 
+[@giacomin:2023a; @giacomin:2023b; @kennedy:2023]. Furthermore, the 
+python interface opens up gyrokinetic analysis to the wide variety of python packages available, 
+allowing for a range of analyses from simple parameter scans to using 
+thousands of linear gyrokinetic runs to develop Gaussian process regression models of the
+linear properties of electromagnetic turbulence [@hornsby:2023]. Pyrokinetics also maintains 
+compatibility with IMAS, a community wide standard for data format [@imbeaux:2015], allowing for
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
compatibility with IMAS, a community-wide standard for data formatting [@imbeaux:2015], allowing for 
```

File: docs/joss/paper.md [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+desirable to benchmark the results of one code against another. Pyrokinetics aims to make this
+easier for researchers by acting as an interface between the different codes, automatically 
+handling the conversion of physical input parameters between different normalisations
+and file formats. Furthermore, gyrokinetics inputs can come from a
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and 
+has been designed to be extensible and easy to add new sources of data. 
+
+The output of gyrokinetic codes is often multidimensional and each code stores this data in a
+different format with different normalisations, and potentially across multiple files. Pyrokinetics will seamlessly read in all this data and
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
different format with different normalisations, potentially across multiple files. Pyrokinetics seamlessly reads in all of this data and
```

File: docs/joss/paper.md [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
stores it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
```

File: docs/joss/paper.md [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+and file formats. Furthermore, gyrokinetics inputs can come from a
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and 
+has been designed to be extensible and easy to add new sources of data. 
+
+The output of gyrokinetic codes is often multidimensional and each code stores this data in a
+different format with different normalisations, and potentially across multiple files. Pyrokinetics will seamlessly read in all this data and
+store it in a single object using an [`xarray`](https://pypi.org/project/xarray/) Dataset, automatically converting the outputs to a 
+standard normalisation (using [`pint`](https://pypi.org/project/Pint/)), allowing for direct comparisons between codes. Furthermore, additional derived
+outputs such as the linear growth rate of a turbulent instability, can be calculated using the exact
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
outputs such as the linear growth rate of a turbulent instability can be calculated using the exact
```

File: docs/joss/paper.md [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+same method such that the modeller is confident that the output is consistent across codes.
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
same method, such that the modeller can be confident that the output is consistent across codes.
```

File: docs/joss/paper.md [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+mitigate turbulent transport and thus requires high levels of confidence in the tools being
+used. Many different gyrokinetic modelling codes are available for use and pyrokinetics aims to 
+standardise the analysis of such simulations.
+
+# Statement of need
+
+[Pyrokinetics](https://github.com/pyro-kinetics/pyrokinetics) is a Python project (package: 
+[`pyrokinetics`](https://pypi.org/project/pyrokinetics/))
+that aims to simplify and standardise gyrokinetic analysis. A wide 
+variety of different gyrokinetic solvers exist that utilise different input file formats and
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
variety of gyrokinetic solvers exist that utilise different input file formats and
```

File: docs/joss/paper.md [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+use different normalisations for plasma parameters such as densities, temperatures, velocities,
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
normalisations for plasma parameters such as densities, temperatures, velocities,
```

File: docs/joss/paper.md [hunk 23]
```diff
@@ -0,0 +1,97 @@
...
+and magnetic fields. To improve confidence in the predictions from gyrokinetic solvers it is often 
+desirable to benchmark the results of one code against another. Pyrokinetics aims to make this
+easier for researchers by acting as an interface between the different codes, automatically 
+handling the conversion of physical input parameters between different normalisations
+and file formats. Furthermore, gyrokinetics inputs can come from a
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
wide variety of modelling tools outside of gyrokinetics, such as X, Y and Z. Pyrokinetics also interfaces with
```

File: docs/joss/paper.md [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
these tools, allowing for the easy generation of both linear and nonlinear gyrokinetic input files, and 
```

File: docs/joss/paper.md [hunk 25]
```diff
@@ -0,0 +1,97 @@
...
+variety of different gyrokinetic solvers exist that utilise different input file formats and
+use different normalisations for plasma parameters such as densities, temperatures, velocities,
+and magnetic fields. To improve confidence in the predictions from gyrokinetic solvers it is often 
+desirable to benchmark the results of one code against another. Pyrokinetics aims to make this
+easier for researchers by acting as an interface between the different codes, automatically 
+handling the conversion of physical input parameters between different normalisations
+and file formats. Furthermore, gyrokinetics inputs can come from a
+wide variety of modelling tools outside of gyrokinetics. Pyrokinetics interfaces with
+these allowing for the easy generation of both linear and nonlinear gyrokinetic input files and 
+has been designed to be extensible and easy to add new sources of data.
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
has been designed to be extensible and simple to incorporate new sources of data. 
```

File: docs/joss/paper.md [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+ - name: General Atomics, San Diego, CA 92121, USA
+   index: 3
+date: 23 August 2023
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Fusion energy offers the potential for a near limitless source of low carbon energy and is often
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
Fusion energy offers the potential for a near limitless source of low-carbon energy and is often 
```

File: docs/joss/paper.md [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+   index: 3
+date: 23 August 2023
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Fusion energy offers the potential for a near limitless source of low carbon energy and is often 
+regarded as a solution for the world's long term energy needs. To realise such a scenario requires
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
regarded as a solution for the world's long-term energy needs. To realise such a scenario requires
```

File: docs/joss/paper.md [hunk 28]
```diff
@@ -0,0 +1,97 @@
...
+the design of high performance fusion reactors capable of maintaining the extreme conditions
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
the design of high-performance fusion reactors capable of maintaining the extreme conditions
```

File: docs/joss/paper.md [hunk 29]
```diff
@@ -0,0 +1,97 @@
...
+necessary to enable fusion. Turbulence is typically the dominant source of transport in fusion
+plasmas, accounting for the majority of the particle and heat loss. Gyrokinetic modelling aims to 
+quantify the level of turbulent transport in fusion reactors and can be used to understand the
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
quantify the level of turbulent transport encountered/expected/observed in fusion reactors and can be used to understand the 
```

File: docs/joss/paper.md [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+# Summary
+
+Fusion energy offers the potential for a near limitless source of low carbon energy and is often 
+regarded as a solution for the world's long term energy needs. To realise such a scenario requires
+the design of high performance fusion reactors capable of maintaining the extreme conditions
+necessary to enable fusion. Turbulence is typically the dominant source of transport in fusion
+plasmas, accounting for the majority of the particle and heat loss. Gyrokinetic modelling aims to 
+quantify the level of turbulent transport in fusion reactors and can be used to understand the 
+major drivers of turbulence. The realisation of fusion critically depends on understanding how to
+mitigate turbulent transport and thus requires high levels of confidence in the tools being
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
mitigate turbulent transport, and thus requires high levels of confidence in the predictive tools being
```

File: docs/joss/paper.md [hunk 31]
```diff
@@ -0,0 +1,97 @@
...
+used. Many different gyrokinetic modelling codes are available for use and pyrokinetics aims to
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
employed. Many different gyrokinetic modelling codes are available and pyrokinetics aims to 
```

File: docs/joss/paper.md [hunk 32]
```diff
@@ -0,0 +1,97 @@
...
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Fusion energy offers the potential for a near limitless source of low carbon energy and is often 
+regarded as a solution for the world's long term energy needs. To realise such a scenario requires
+the design of high performance fusion reactors capable of maintaining the extreme conditions
+necessary to enable fusion. Turbulence is typically the dominant source of transport in fusion
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
necessary to enable fusion. Turbulence is typically the dominant source of transport in magnetically-confined fusion
```

File: docs/joss/paper.md [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+plasmas, accounting for the majority of the particle and heat loss. Gyrokinetic modelling aims to
```
Comment (hgd511) [inline docs/joss/paper.md]:
```suggestion
plasmas, accounting for the majority of the particle and heat losses. Gyrokinetic modelling aims to 
```

File: docs/joss/paper.md (hunk 29 above)
Comment (bpatel2107) [reply docs/joss/paper.md]: Encountered seems best here

**Comment by arkabokshi** [comment]
- add tag
- standardize gyrokinetic(s) inputs/solvers/modellers/runs/codes etc
- capitalize Pyrokinetics to be consistent

File: docs/joss/paper.md [hunk 34]
```diff
@@ -69,7 +70,7 @@ quantify the level of turbulent transport encountered in fusion reactors and can
 major drivers of turbulence. The realisation of fusion critically depends on understanding how to
 mitigate turbulent transport, and thus requires high levels of confidence in the predictive tools being
 employed. Many different gyrokinetic modelling codes are available and Pyrokinetics aims to
-standardise the analysis of such simulations.
+standardise the analysis of such computationally demanding simulations.
```
Comment (bpatel2107) [inline docs/joss/paper.md:80]: Not always true. Pyro works with TGLF and there's no reason you can't run low cost simulations like GA-STD/CBC which don't need to be very expensive, especially when linear

Comment (arkabokshi) [reply docs/joss/paper.md:80]: Agreed - was trying to emphasize that standardization / accurate inputs/conversions is especially important given runs are particularly challenging!

Comment (arkabokshi) [reply docs/joss/paper.md:80]: Would it be good to change https://github.com/pyro-kinetics/pyrokinetics/blob/038eb7aee01ce05e5b2fa74334048dc74f2ab57e/README.md?plain=1#L180 to something like Used and Developed by to capture the institutions of all the contributing author?

Comment (bpatel2107) [reply docs/joss/paper.md:80]: Happy to add institutions but for all the authors we have the citation file

**Comment by bpatel2107** [comment]
@juanruizruiz @tomneiser this paper is ready to go. You've contributed quite a bit albeit not in the unstable branch. Please add yourself to the list of authors at the end. Currently unable to do it easily whilst travelling!

**Pull request merged**